tensorflow_addons/losses/kappa_loss.py

```diff
@@ -0,0 +1,175 @@
+
+    denominator = tf.reduce_sum(
+        tf.matmul(
+            tf.reduce_sum(y_true, axis=0, keepdims=True),
+            tf.matmul(
+                weight_mat,
+                tf.reduce_sum(y_pred, axis=0, keepdims=True),
+                transpose_b=True,
+            ),
+        )
+    )
+    denominator /= tf.cast(tf.shape(y_true)[0], dtype=dtype)
+    return tf.math.log(tf.math.divide_no_nan(numerator, denominator) + epsilon)
+
+
+@tf.keras.utils.register_keras_serializable(package="Addons")
+class WeightedKappaLoss(tf.keras.losses.Loss):
+    """Implements the Weighted Kappa loss function.
+
+    Weighted Kappa loss was introduced in the
+    [Weighted kappa loss function for multi-class classification
+    of ordinal data in deep learning]
+    (https://www.sciencedirect.com/science/article/abs/pii/S0167865517301666).
+    Weighted Kappa is widely used in Ordinal Classification Problems.
+    The loss value lies in [-inf, log 2], where log 2
+     means the random prediction.
+
+    Usage:
+
+    ```python
+    kappa_loss = WeightedKappaLoss(num_classes=4)
+    y_true = tf.constant([[0, 0, 1, 0], [0, 1, 0, 0],
+                          [1, 0, 0, 0], [0, 0, 0, 1]])
+    y_pred = tf.constant([[0.1, 0.2, 0.6, 0.1], [0.1, 0.5, 0.3, 0.1],
+                          [0.8, 0.05, 0.05, 0.1], [0.01, 0.09, 0.1, 0.8]])
+    loss = kappa_loss(y_true, y_pred)
+    print('Loss: ', loss.numpy())  # Loss: -1.1611923
+    ```
+
+    Usage with `tf.keras` API:
+    ```python
+    # outputs should be softmax results
+    # if you want to weight the samples, just multiply the outputs
+    # by the sample weight.
+    model = tf.keras.Model(inputs, outputs)
+    model.compile('sgd', loss=tfa.losses.WeightedKappa(num_classes=4))
+    ```
+    """
+
+    @typechecked
+    def __init__(
+        self,
+        num_classes: int,
+        weightage: Optional[str] = "quadratic",
+        name: Optional[str] = "cohen_kappa_loss",
+        epsilon: Optional[Number] = 1e-6,
+        dtype: Optional[tf.DType] = tf.float32,
+    ):
+        """Creates a `WeightedKappa` instance.
+
+        Args:
+          num_classes: Number of unique classes in your dataset.
+          weightage: (Optional) Weighting to be considered for calculating
+            kappa statistics. A valid value is one of
+            ['linear', 'quadratic']. Defaults to `quadratic` since it's
+            mostly used.
+          name: (Optional) String name of the metric instance.
+          epsilon: (Optional) increment to avoid log zero,
+            so the loss will be log(1 - k + epsilon), where k belongs to
+            [-1, 1], usually you can use the default value which is 1e-6.
+          dtype: (Optional) Data type of the metric result.
+            Defaults to `tf.float32`.
+        Raises:
+          ValueError: If the value passed for `weightage` is invalid
+            i.e. not any one of ['linear', 'quadratic']
+        """
+
+        super(WeightedKappaLoss, self).__init__(
```

Comment:
```suggestion
        super().__init__(
```

Arguments in `super` are only needed in python 2.